pkgs/build-support/cc-wrapper/cc-wrapper.sh

```diff
@@ -127,11 +125,16 @@ if [ "$NIX_@infixSalt@_ENFORCE_NO_NATIVE" = 1 ]; then
     params=(${rest+"${rest[@]}"})
 fi
 
+if [[ "$stdInclude" = 1 ]]; then
+    NIX_@infixSalt@_CFLAGS_COMPILE="$NIX_@infixSalt@_CSTDLIB_COMPILE $NIX_@infixSalt@_CFLAGS_COMPILE"
+fi
+NIX_@infixSalt@_CFLAGS_LINK="$NIX_@infixSalt@_CSTDLIB_LINK  $NIX_@infixSalt@_CFLAGS_LINK"
+
 if [[ "$isCpp" = 1 ]]; then
-    if [[ "$cppInclude" = 1 ]]; then
-        NIX_@infixSalt@_CFLAGS_COMPILE+=" ${NIX_@infixSalt@_CXXSTDLIB_COMPILE:-@default_cxx_stdlib_compile@}"
+    if [[ "$stdInclude" = 1 ]]; then
+        NIX_@infixSalt@_CFLAGS_COMPILE="$NIX_@infixSalt@_CXXSTDLIB_COMPILE $NIX_@infixSalt@_CFLAGS_COMPILE"
```

Comment:
Does the order matter here? I recall something about `+=` being much faster.

Reply:
Is `+=` much faster? If so, we probably should rewrite everything to use it.

Reply:
The order does sometimes matter: wrapper-provided things should generally be overrideable by explicit arguments.

What I currently have causes a *segfault* during bootstrapping, so we are far from being able to see what `+=` we can get away with, sadly.

Reply:
```console
$ hyperfine "bash -c 'a=0; for i in \$(seq 1 100000); do a+=0; done'" "bash -c 'a=0; for i in \$(seq 1 100000); do a=\"0$a\"; done'"
Benchmark #1: bash -c 'a=0; for i in $(seq 1 100000); do a+=0; done'
  Time (mean ± σ):     983.5 ms ±   8.9 ms    [User: 494.8 ms, System: 491.3 ms]
  Range (min … max):   973.2 ms … 999.2 ms    10 runs

Benchmark #2: bash -c 'a=0; for i in $(seq 1 100000); do a="0"; done'
  Time (mean ± σ):     130.6 ms ±   2.3 ms    [User: 123.1 ms, System: 10.3 ms]
  Range (min … max):   127.3 ms … 135.8 ms    22 runs

Summary
  'bash -c 'a=0; for i in $(seq 1 100000); do a="0"; done'' ran
    7.53 ± 0.15 times faster than 'bash -c 'a=0; for i in $(seq 1 100000); do a+=0; done''
hyperfine "bash -c 'a=0; for i in \$(seq 1 100000); do a+=0; done'"   7.96s user 5.32s system 100% cpu 13.208 total
```

Reply:
This is converted back to the way it was.